### PR TITLE
Outline for custom reader post

### PR DESCRIPTION
**Pull request by richardjgowers**
I'm not sure if this is suitable for a blog post, it's a little more technical and hacky than others, but I think it's a nice illustration of how you can hack MDAnalysis a little to make it work for weird formats (without requiring you to build it from source).

Obviously this needs fleshing out with more explanations, but I think the basic content (in terms of code blocks) I wanted to show is here.

Thought @MDAnalysis/coredevs ?

### CODE REVIEW

**Comment by kain88-de** [comment]
I like the idea. I'll have a thorough look at it when I'm home. A link to the general reader docs for more complicated formats at the end might be nice.

**Comment by jbarnoud** [comment]
I like the idea too. It has completely its place on the blog!

File: _posts/2018-04-15-customreaders.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+H blue     1.0 1.5 1.5
+H blue     1.0 0.5 0.5
+C red      2.0 2.0 2.0
+H yellow   2.0 2.5 2.5
+H yellow   2.0 1.5 1.5
+O octarine 3.0 3.0 3.0
+"""
+with open('myfile.xyzc', 'w') as out:
+    out.write(myfile)
+```
```
Comment (kain88-de) [inline _posts/2018-04-15-customreaders.md]: You can use liquid syntax to have correct syntax highlight preview on a local build

{% highlight python %}
...
{% endhighlight %}

File: _posts/2018-04-15-customreaders.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        372 
+        373     def _reopen(self):
+
+
+    EOFError: could not convert string to float: red
+
+
+Here, as expected, the XYZReader chokes on the word 'red' where it expects a floating point number.
+
+We can rewrite the XYZReader to anticipate the extra column:
```
Comment (kain88-de) [inline _posts/2018-04-15-customreaders.md]: We can create a new reader based on the XYZReader that anticipates the extra column

File: _posts/2018-04-15-customreaders.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+            ts = self.ts
+
+        f = self.xyzfile
+
+        try:
+            # we assume that there are only two header lines per frame
+            f.readline()
+            f.readline()
+            for i in range(self.n_atoms):
+                self.ts._pos[i] = np.float32(f.readline().split()[2:5])
```
Comment (kain88-de) [inline _posts/2018-04-15-customreaders.md]: I thought that solution was slow ;-)

File: _posts/2018-04-15-customreaders.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+            ts.frame += 1
+            return ts
+        except (ValueError, IndexError) as err:
+            raise EOFError(err)
+            
+class XYZCParser(XYZParser):
```
Comment (kain88-de) [inline _posts/2018-04-15-customreaders.md]: I wouldn't have this here and yet. You haven't talked about the topology parsing yet. I think it's better to show an error and say that we can either give a topology file mdanalysis understands or this short class to add a xyzc topology format.

File: _posts/2018-04-15-customreaders.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    [[ 1.   1.   1. ]
+     [ 1.   1.5  1.5]
+     [ 1.   0.5  0.5]
+     [ 2.   2.   2. ]
+     [ 2.   2.5  2.5]
+     [ 2.   1.5  1.5]
+     [ 3.   3.   3. ]]
+
+
+Now it seems to work.
```
Comment (kain88-de) [inline _posts/2018-04-15-customreaders.md]: I don't like the work `seems` here. It indicates we aren't sure. But we should be sure that it works at this points.

File: _posts/2018-04-15-customreaders.md [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+But what about our colour field?
```
Comment (kain88-de) [inline _posts/2018-04-15-customreaders.md]: Show a example that the attribute doesn't exist.  Then

`To add the color field to our topology we need to teach MDAnalysis about the color attribute in general and our new XYZC parser to read it.`

File: _posts/2018-04-15-customreaders.md [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+        return top
+```
+
+
+```python
+u = mda.Universe('myfile.xyzc')
+```
+
+
+```python
```
Comment (kain88-de) [inline _posts/2018-04-15-customreaders.md]: last one can be one code-field

File: _posts/2018-04-15-customreaders.md [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+print u.atoms.colours
+print u.atoms.complimentary
+```
+
+    ['red' 'blue' 'blue' 'red' 'yellow' 'yellow' 'octarine']
+    ['green' 'orange' 'orange' 'green' 'purple' 'purple' '???']
+
```
Comment (kain88-de) [inline _posts/2018-04-15-customreaders.md]: Now we can access the colors as epxected.

**Comment by hmacdope** [comment]
@richardjgowers stale? Reopen if not.

**Pull request closed without merging**